### ISSUE TITLE
Feature request: Diagram-from-text support for instance through mermaidjs

### ISSUE THREAD

**Issue by em-** [hotdoc/hotdoc#142]
It would be extremely nice to be able to input flowcharts/graphs/diagrams as text in the document and have something like [mermaidjs](https://mermaidjs.github.io/) to render them.

**Comment by MathieuDuponchelle** [comment]
Great idea, how would you go about exposing this though? :)

**Comment by MathieuDuponchelle** [comment]
@em- , I suppose we would take the same approach as Markdown Plus (https://mdp.tylingsoft.com/), except I'd rather not have the language hint of code blocks intended to be rendered with mermaid be `mermaid`, but something more qualified such as `render-mermaid`, in order not to interfere with standard syntax highlighting if prism ever got a mermaid highlighter. Do you know if mermaid exposes a python API?

**Comment by MathieuDuponchelle** [comment]
Other option is client-side rendering of course

**Comment by em-** [comment]
Yep, all the [examples](https://github.com/geekplux/markvis) [in the](https://support.typora.io/Draw-Diagrams-With-Markdown/) [wild](https://www.npmjs.com/package/markdown-it-charts) I could find use the tagged code-block escape hatch. I also like the `render-mermaid` bit. My concern about client-side rendering is for PDF generation, which is something I still care about.

**Comment by MathieuDuponchelle** [comment]
hotdoc is purely a markdown -> html tool for this reason (amongst others). The path to pdf is through html, this has been tested in the past and gave acceptable results :) I think client-side rendering should be all right, if you want inspiration for implementing this I think a mermaid extension would be pretty similar to the [syntax highlighting extension](https://github.com/hotdoc/hotdoc/blob/master/hotdoc/extensions/syntax_highlighting/syntax_extension.py) :)
